src/frequenz/client/electricity_trading/_client.py

```diff
@@ -123,7 +125,11 @@ def __init__(
         if not hasattr(
             self, "_initialized"
         ):  # Prevent re-initialization of existing instances
-            super().__init__(server_url, ElectricityTradingServiceStub, connect=connect)
+            super().__init__(server_url, connect=connect)
+            self._stub = cast(
+                ElectricityTradingServiceAsyncStub,
+                ElectricityTradingServiceStub(self.channel),
+            )
```

Comment:
It seems there's another problem here.

We have to create the `_stub` when we are connecting.  So if `connect=False`, we can't create the stub immediately, because there's no channel.

And we're going to have to override `connect`, to call super.connect(), then create the stub.

@llucax this is also not covered in the upgrade notes.

I'll try to make a working patch for this now.